src/PoshArmDeployment/Public/ResourceProviders/Microsoft.Network/ApplicationGateway/Add-ArmApplicationGatewayHealthProbe.ps1

```diff
@@ -0,0 +1,55 @@
+function Add-ArmApplicationGatewayHealthProbe {
+    [CmdletBinding(SupportsShouldProcess = $true)]
+    [OutputType("ApplicationGateway")]
+    Param(
+        [PSTypeName("ApplicationGateway")]
+        [Parameter(Mandatory, ValueFromPipeline)]
+        $ApplicationGateway,
+        [ValidatePattern('^(\[.*\]|[a-zA-Z0-9-_.]*)$')]
+        [string]
+        $Name = "default",
+        [string]
+        [ValidateSet("Http", "Https")]
+        $Protocol = "Http",
+        [string]
+        $HostName,
```

Comment:
mandatory?
in fact it depends on the PickHostFromBackend.... so we should probably create different ParameterSet for those 2 cases